test/index.ts

```diff
@@ -134,4 +172,45 @@ describe('Mutation', () => {
       expect(await getNumberOfUsersInDB()).to.equal(1);
     });
   });
+
+  describe('login', () => {
+    beforeEach(async () => {
+      await userRepository.delete({});
+    });
+
+    it('should succeed if credentials are correct', async () => {
+      const newUser = await insertUserInDB(defaultLoginInput);
+
+      const response = (await requestLogin(defaultLoginInput)).data;
+      const loginInfo = response.data.login;
+
+      expect(loginInfo.user).excluding(['password', 'id']).to.deep.equal(newUser);
+      expect(+loginInfo.user.id).to.deep.equal(newUser.id);
+      expect(loginInfo).to.haveOwnProperty('token');
+    });
+
+    it('should fail if no user has the given email', async () => {
+      await insertUserInDB(defaultLoginInput);
+
+      const response = (await requestLogin({ ...defaultLoginInput, email: 'wrong@email.com' })).data;
+
+      expect(response.errors[0]).excluding('stacktrace').to.deep.equal({
+        message: 'Invalid credentials',
+        code: 400,
+        details: 'No user with this email',
+      });
+    });
+
+    it('should fail if password is wrong', async () => {
+      await insertUserInDB(defaultLoginInput);
+
+      const response = (await requestLogin({ ...defaultLoginInput, password: 'wr0ngpassword' })).data;
+
+      expect(response.errors[0]).excluding('stacktrace').to.deep.equal({
+        message: 'Invalid credentials',
+        code: 400,
+        details: 'Wrong password',
+      });
+    });
+  });
```

Comment:
Talvez um caso de teste pra e-mail sem formato de e-mail?

Reply:
Eu não fiz validação do e-mail no caso do login (só no `createUser`). Isso é necessário? Se não for, eu ainda assim preciso fazer esse teste?

Reply:
Necessário não é pq ele vai cair no not-found. É mais um nice-to-have pq assim vc n faz uma busca no banco sabendo que já vai dar errado.